src/cli/index.ts

```diff
@@ -0,0 +1,306 @@
+    `${chalk.bold("Importing")} ${inputType} project from path ${chalk.white(fullInputPath)}.`,
+    async () => {
+      let file: Buffer;
+      try {
+        file = await fs.readFile(fullInputPath);
+      } catch (err) {
+        if (err instanceof Object && "code" in err) {
+          switch (err.code) {
+            case "ENOENT":
+              throw new StepError("File not found.");
+            case "EISDIR":
+              throw new StepError("Input path is a directory but should be a file.");
+          }
+        }
+        throw err;
+      }
+
+      let project: Project;
+      switch (inputType) {
+        case "sb3": {
+          project = await Project.fromSb3(file);
+          break;
+        }
+      }
+      return project;
+    }
+  );
+
+  switch (outputType) {
+    case "leopard": {
+      const leopard = await writeStep(`${chalk.bold("Converting")} project to ${chalk.white("Leopard")}.`, () => {
+        return project.toLeopard();
+      });
+
+      const fullOutputPath = path.resolve(process.cwd(), output);
+
+      await writeStep(`${chalk.bold("Exporting")} project to directory ${chalk.white(fullOutputPath)}.`, async () => {
+        // First, make sure the output path is an empty directory (create it if it doesn't exist)
+        let existingFiles: string[];
+        try {
+          existingFiles = await fs.readdir(fullOutputPath);
+        } catch (err) {
+          if (err instanceof Object && "code" in err) {
+            switch (err.code) {
+              case "ENOENT":
+                // Directory does not exist, create it
+                await fs.mkdir(output, { recursive: true });
+                existingFiles = [];
+                break;
+              case "ENOTDIR":
+                throw new StepError("Output path is a file, not a directory.");
+              default:
+                throw err;
+            }
+          } else {
+            throw err;
+          }
+        }
+
+        if (existingFiles.length > 0) {
+          throw new StepError("Output directory is not empty.");
+        }
+
+        // Write the files
+        for (const [filename, contents] of Object.entries(leopard)) {
+          const filePath = path.resolve(fullOutputPath, filename);
+
+          // Create directories as needed
+          await fs.mkdir(path.dirname(filePath), { recursive: true });
+
+          // Write the file
+          await fs.writeFile(filePath, contents);
+        }
+
+        for (const target of [project.stage, ...project.sprites]) {
+          const costumeDir = path.join(target.name, "costumes");
+          for (const costume of target.costumes) {
+            const filename = path.join(costumeDir, `${costume.name}.${costume.ext}`);
+            const asset = Buffer.from(costume.asset as ArrayBuffer);
+            await fs.mkdir(path.resolve(fullOutputPath, filename, ".."), { recursive: true });
+            await fs.writeFile(path.resolve(fullOutputPath, filename), asset);
```

Comment:
This is *almost* exactly on point, but let's move the `fs.mkdir` call out of the for loop here—we only need to create it once for each sprite, not once for each costume in each sprite.

```diff
  const costumeDir = path.join(target.name, "costumes");
+await fs.mkdir(path.resolve(fullOutputPath, costumeDir), { recursive: true });
  for (const costume of target.costumes) {
    const filename = path.join(costumeDir, `${costume.name}.${costume.ext}`);
    const asset = Buffer.from(costume.asset as ArrayBuffer);
-   await fs.mkdir(path.resolve(fullOutputPath, filename, ".."), { recursive: true });
    await fs.writeFile(path.resolve(fullOutputPath, filename), asset);
```

Same thing for sounds below, of course.

Reply:
Good call. Thank you